examples/data_view/row_table_example.py

```diff
@@ -0,0 +1,123 @@
+# (C) Copyright 2005-2020 Enthought, Inc., Austin, TX
+# All rights reserved.
+#
+# This software is provided without warranty under the terms of the BSD
+# license included in LICENSE.txt and may be redistributed only under
+# the conditions described in the aforementioned license. The license
+# is also available online at http://www.enthought.com/licenses/BSD.txt
+#
+# Thanks for using Enthought open source!
+
+from traits.api import HasStrictTraits, Instance, Int, Str, List
+
+from pyface.api import ApplicationWindow, GUI
+from pyface.data_view.data_models.data_accessors import AttributeDataAccessor
+from pyface.data_view.data_models.row_table_data_model import (
+    RowTableDataModel
+)
+from pyface.data_view.data_view_widget import DataViewWidget
+from pyface.data_view.i_data_view_widget import IDataViewWidget
+from pyface.data_view.value_types.api import ConstantValue, IntValue, TextValue
+
+from example_data import any_name, family_name, age, street, city, country
+
+
+# The data model
+
+class Address(HasStrictTraits):
+
+    street = Str()
+
+    city = Str()
+
+    country = Str()
+
+
+class Person(HasStrictTraits):
+
+    name = Str()
+
+    age = Int()
+
+    address = Instance(Address, ())
+
+
+row_header_data = AttributeDataAccessor(
+    title='People',
+    attr='name',
+    value_type=TextValue(),
+)
+
+column_data = [
+    AttributeDataAccessor(
+        attr="age",
+        value_type=IntValue(minimum=0),
+    ),
+    AttributeDataAccessor(
+        attr="address.street",
+        value_type=TextValue(),
+    ),
+    AttributeDataAccessor(
+        attr="address.city",
+        value_type=TextValue(),
+    ),
+    AttributeDataAccessor(
+        attr="address.country",
+        value_type=TextValue(),
+    ),
+]
+
+
+class MainWindow(ApplicationWindow):
+    """ The main application window. """
+
+    #: A collection of People.
+    data = List(Instance(Person))
```

Comment:
It is not a use case for this example, but if this data is to be mutated, we will need to observe mutations on this list as well.